lib/jxl/encode.cc

```diff
@@ -418,6 +427,24 @@ void JxlEncoderInitBasicInfo(JxlBasicInfo* info) {
   info->animation.have_timecodes = JXL_FALSE;
 }
 
+void JxlEncoderInitFrameHeader(JxlFrameHeader* info) {
+  info->duration = 0;
+  info->timecode = 0;
+  info->name_length = 0;
+  info->is_last = JXL_FALSE;
+  info->layer_info.crop_x0 = 0;
+  info->layer_info.crop_y0 = 0;
+  // TODO(lode): must these be set to image size instead, or will caller do
+  // this?
```

Comment:
Initializing to image dimensions seems a more sensible default value to me, so you don't have to explicitly set these if you are sending full-sized frames or layers.

Reply:
See other comment: After thinking about the options (see other comment) I have now chosen to follow the specification default, and that has it at 0. User should set them if they want to use crop. We should also add have_crop, but not in this MR since blending (of which cropping is part) is not for this MR